Meridian59.DebugUI/ObjectEditor.cs

```diff
@@ -0,0 +1,22 @@
+﻿using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Data;
+using System.Drawing;
+using System.Linq;
+using System.Text;
+using System.Windows.Forms;
+using Meridian59.Data.Models.AdminData;
+
+namespace Meridian59.DebugUI
+{
+    public partial class ObjectEditor : Form
+    {
+        public ObjectEditor(AdminObject obj)
```

Comment:
Please do not pass the datasource as a constructor parameter.

Instead define a property "DataSource" on your UI (view), according to MVC pattern the type of this property of course would be the one of the according underlying model providing the data for the view (your "AdminData" class for example if you need all of it)

Please have a look at "RoomInfoView.cs" or "RoomObjectsView.cs" for examples for "DataSource" property.
You can adopt their definition of a "DataSource" property. Don't forget the [DesignerSerialization...] attributes above the signature.